sensor/kubernetes/listener/resources/secrets.go

```diff
@@ -290,7 +290,9 @@ func (s *secretDispatcher) processDockerConfigEvent(secret, oldSecret *v1.Secret
 			}
 
 			// The secrets captured in this block are used for delegated image scanning.
-			if env.LocalImageScanningEnabled.BooleanSetting() && !env.DelegatedScanningDisabled.BooleanSetting() {
+			if env.LocalImageScanningEnabled.BooleanSetting() &&
+				!env.DelegatedScanningDisabled.BooleanSetting() &&
+				action != central.ResourceAction_REMOVE_RESOURCE {
```

Comment:
Should we consider removing from the store if this is a remove action?

Reply:
We cannot do that reliably current state because we store the secrets by endpoint - many secrets in a namespace may refer to the same endpoint. This will be addressed when/if ROX-16821 (use pull secrets by name) is prioritized.